test/lib/TestDialect/TestDialect.cpp

```diff
@@ -41,6 +41,19 @@ struct TestOpAsmInterface : public OpAsmDialectInterface {
     if (auto asmOp = dyn_cast<AsmDialectInterfaceOp>(op))
       setNameFn(asmOp, "result");
   }
+
+  void getAsmBlockArgumentNames(Block *block,
+                                OpAsmSetValueNameFn setNameFn) const final {
+    auto op = block->getParentOp();
+    if (auto arrayAttr = op->getAttrOfType<ArrayAttr>("arg_names")) {
+      auto args = block->getArguments();
+      for (unsigned i = 0; i < std::min(arrayAttr.size(), args.size()); i++) {
```

Comment:
Cache the end iterator to avoid recomputation, and prefer preincrement

e = std::min(...); i != e; ++i

Reply:
Why?

Reply:
This is part of the MLIR(LLVM) coding standards:
https://llvm.org/docs/CodingStandards.html